### PR TITLE
Add borderRadius for the bottom nav bar

### PR DESCRIPTION
**Pull request by vinothvino42**
Ticket: [#869](https://github.com/EnsembleUI/ensemble/issues/869)

```yaml
BottomNavBar:
  styles:
    {
      backgroundColor: indigo,
      color: green,
      selectedColor: blue,
      floatingBackgroundColor: blue,
      floatingIconColor: white,
      notchColor: white,
      height: 60,
      padding: 0 20,
      borderRadius: 30 30 0 0,
    }
```

### CODE REVIEW

File: lib/framework/view/bottom_nav_page_group.dart [hunk 1]
```diff
@@ -232,20 +234,24 @@ class _BottomNavPageGroupState extends State<BottomNavPageGroup>
...
-      padding: widget.menu.styles?['padding'],
-      color: unselectedColor,
-      selectedColor: selectedColor,
-      notchedShape: const CircularNotchedRectangle(),
-      onTabSelected: controller.jumpToPage,
-      items: navItems,
-      isFloating: fabMenuItem != null,
-      floatingAlignment: floatingAlignment,
-      floatingMargin: floatingMargin,
+    return ClipRRect(
```
Comment (vusters) [inline lib/framework/view/bottom_nav_page_group.dart]: expensive. Don't use ClipRRect unles borderRadius is specified.

Comment (vusters) [inline lib/framework/view/bottom_nav_page_group.dart]: There'll be requirements for margin. Why don't we wrap it inside a Container so we can add margin in addition to border radius.

Comment (vinothvino42) [reply lib/framework/view/bottom_nav_page_group.dart]: @vusters Can you review it again? I changed it to Container and added `margin` and refactored the `padding`, `borderRadius` properties

File: lib/framework/view/bottom_nav_page_group.dart [hunk 2]
```diff
@@ -365,20 +368,21 @@ class EnsembleBottomAppBarState extends State<EnsembleBottomAppBar> {
...
     return Theme(
       data: ThemeData(useMaterial3: false),
-      child: ClipRRect(
-        borderRadius: widget.borderRadius ?? BorderRadius.zero,
+      child: Container(
+        margin: Utils.optionalInsets(widget.margin) ?? EdgeInsets.zero,
+        decoration: BoxDecoration(
+          borderRadius: widget.borderRadius ?? BorderRadius.zero,
+        ),
+        clipBehavior: Clip.hardEdge,
```
Comment (vusters) [inline lib/framework/view/bottom_nav_page_group.dart:376]: use Clip.none if border radius is not specified. Minimizing clip if possible if we dn't need it.

**Review by vusters — APPROVED** [review]

**Pull request merged**